lib/ClangImporter/ImportName.cpp

```diff
@@ -1729,9 +1729,16 @@ ImportedName NameImporter::importNameImpl(const clang::NamedDecl *D,
 
 /// Returns true if it is expected that the macro is ignored.
 static bool shouldIgnoreMacro(StringRef name, const clang::MacroInfo *macro) {
-  // Ignore include guards.
-  if (macro->isUsedForHeaderGuard())
-    return true;
+  // Ignore include guards. Try not to ignore definitions of useful constants,
+  // which may end up looking like include guards.
+  if (macro->isUsedForHeaderGuard()
+      && macro->getNumTokens() == 1
+      && macro->tokens()[0].isLiteral()) {
+      auto* tok = macro->tokens()[0].getLiteralData();
+      if (tok && strlen(tok) == 1 && strncmp("1", tok, 1) == 0) {
```

Comment:
LLVM has a very useful type, `StringRef`, that makes working with strings a lot easier. So this code can be written `if (tok && StringRef(tok) == "1")`